### PR TITLE
network: add RouteMetric= setting in [Address] and more

### LINKED ISSUE #928: systemd-networkd: make sure RouterMetric= also applies to routes to the local subnet
I have a wireless and an ethernet adaptor. If the ethernet is connected, I don't want the wireless to connect, even if there's a known AP in range. There is currently no way to configure this in systemd-networkd and I have to use an alternative like networkmanager.

### LINKED ISSUE #18496: networkd: allow fine-tune prefix route for DHCPv4 assigned address
**Is your feature request related to a problem? Please describe.**
IPv4 prefix routes generated by kernel can be disabled by `AddPrefixRoute=no`.
If anyone wants to fine-tune the prefix routes (e.g. `Table=`, `Metric=`), standalone `[Route]=` sections are required.
This is problematic for DHCPv4 networks where IP addresses are leased from undetermined subnets (multiple address pools) and `Destination=` cannot be determined before DHCPv4 lease.

#928 might be related.

**Describe the solution you'd like**
```
[Route]
Destination=_dhcp4_prefix
Scope=link
Table=1
Metric=1
```
For any DHCPv4 assigned address, a prefix route is configured using this template.

**The systemd version you checked that didn't have the feature you are asking for**
main branch

### LINKED ISSUE #19028: Allow controlling of prefix route metric in [address] and [dhcp*]
**Is your feature request related to a problem? Please describe.**

There is no easy way to control the metric of prefix routes defined via [address] or [dhcp*] sections

**Describe the solution you'd like**

Similarly to how we can define metric in [Route] section, it would be nice to be able to do that at [address]/[[dhcp*] level, so the equivalent of the following is possible in systemd:

`ip ad add 192.168.50.50/24 dev eno1 metric 123`

which normally results with a prefix route being added with required metric:

```
ip ro sh
192.168.50.0/24 dev eno1 proto kernel scope link src 192.168.50.50 metric 123
```

In context of dhcp, the already present RouteMetric option could be taken into account, as it currently isn't, for example:

```
default via 10.13.36.1 dev eno1 proto dhcp src 10.13.37.254 metric 200 
10.13.36.0/23 dev eno1 proto kernel scope link src 10.13.37.254 
10.13.36.1 dev eno1 proto dhcp scope link src 10.13.37.254 metric 200 
```

Note missing metric on prefix route. This is generated by the following .network file config:

```
[DHCP]
UseDNS=true
UseDomains=true
UseRoutes=true
RouteMetric=200
```

**Describe alternatives you've considered**

The only workaround is to block automatic prefix route creation, and create it independently via [route] section, but that's messy. The dhcp case seems impossible to control.

**The systemd version you checked that didn't have the feature you are asking for**

systemd 247 (247.4-2-arch), also current origin/master as of writing of this RFE.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #928.
Closes #18496.
Closes #19028.

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -1099,6 +1099,13 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
             Defaults to <literal>global</literal>.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>RouteMetric=</varname></term>
+          <listitem>
+            <para>The metric of the prefix route. Takes an unsigned integer in the range 0…4294967295.
```
Comment (poettering) [inline man/systemd.network.xml]: maybe explain what the prefix route is, might not be a term everyone groks, i.e. add "(the route pointing to the subnet of the configured IP addresses', taking the configured subnet mask into account)" or someting like that

File: man/systemd.network.xml [hunk 2]
```diff
@@ -2158,6 +2158,14 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
           <para>As in the [Address] section, but defaults to true.</para>
         </listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>RouteMetric=</varname></term>
+        <listitem>
+          <para>The metric of the route. Takes an unsigned integer in the range 0…4294967295. When
```
Comment (poettering) [inline man/systemd.network.xml]: "the route"? should probably say "the route to the delegated prefix subnet" or so

**Review by poettering — COMMENTED** [review]
lgtm, but i think the docs could be slightly improved to say which route's metrics are specifically meant, i.e. the fact that these sections all synthesize routes is not totally obvious I think, and that hence deserves some minimal explanation

**Comment by yuwata** [comment]
@poettering Thank you for the comments. The above points are addressed. Upgrading the green label.

**Pull request merged**